jobbergate-api/jobbergate_api/apps/job_submissions/schemas.py

```diff
@@ -168,64 +178,72 @@ class JobProperties(BaseModel, extra=Extra.forbid):
         ge=0,
         le=1,
     )
-    gres: Optional[str] = Field(
+    gres: Optional[LengthLimitedStr] = Field(
         description="Specifies a comma delimited list of generic consumable resources."
     )
     gres_flags: Optional[Literal["disable-binding", "enforce-binding"]] = Field(
         description="Specify generic resource task binding options."
     )
-    gpu_binding: Optional[str] = Field(description="Requested binding of tasks to GPU.")
-    gpu_frequency: Optional[str] = Field(description="Requested GPU frequency.")
-    gpus: Optional[str] = Field(description="GPUs per job.")
-    gpus_per_node: Optional[str] = Field(description="GPUs per node.")
-    gpus_per_socket: Optional[str] = Field(description="GPUs per socket.")
-    gpus_per_task: Optional[str] = Field(description="GPUs per task.")
+    gpu_binding: Optional[LengthLimitedStr] = Field(description="Requested binding of tasks to GPU.")
+    gpu_frequency: Optional[LengthLimitedStr] = Field(description="Requested GPU frequency.")
+    gpus: Optional[LengthLimitedStr] = Field(description="GPUs per job.")
+    gpus_per_node: Optional[LengthLimitedStr] = Field(description="GPUs per node.")
+    gpus_per_socket: Optional[LengthLimitedStr] = Field(description="GPUs per socket.")
+    gpus_per_task: Optional[LengthLimitedStr] = Field(description="GPUs per task.")
     hold: Optional[bool] = Field(
         description="Specify the job is to be submitted in a held state (priority of zero)."
     )
     kill_on_invalid_dependency: Optional[bool] = Field(
         description="If a job has an invalid dependency, then Slurm is to terminate it."
     )
-    licenses: Optional[str] = Field(
+    licenses: Optional[LengthLimitedStr] = Field(
         description=(
             "Specification of licenses (or other resources available on all nodes of the cluster) "
             "which must be allocated to this job."
         )
     )
-    mail_type: Optional[str] = Field(description="Notify user by email when certain event types occur.")
-    mail_user: Optional[str] = Field(
+    mail_type: Optional[LengthLimitedStr] = Field(
+        description="Notify user by email when certain event types occur."
+    )
+    mail_user: Optional[LengthLimitedStr] = Field(
         description="User to receive email notification of state changes as defined by mail_type."
     )
-    mcs_label: Optional[str] = Field(description="This parameter is a group among the groups of the user.")
-    memory_binding: Optional[str] = Field(description="Bind tasks to memory.")
+    mcs_label: Optional[LengthLimitedStr] = Field(
+        description="This parameter is a group among the groups of the user."
+    )
+    memory_binding: Optional[LengthLimitedStr] = Field(description="Bind tasks to memory.")
     memory_per_cpu: Optional[int] = Field(description="Minimum real memory per cpu (MB).")
     memory_per_gpu: Optional[int] = Field(description="Minimum memory required per allocated GPU.")
```

Comment:
Should this be NonNegativeInt?

```suggestion
    memory_per_cpu: Optional[NonNegativeInt] = Field(description="Minimum real memory per cpu (MB).")
    memory_per_gpu: Optional[NonNegativeInt] = Field(description="Minimum memory required per allocated GPU.")
```